core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/WebAppsDefaultHeartbeatProvider.java

```diff
@@ -0,0 +1,136 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * <h1>WebApp Heartbeat Property Provider</h1>
+ * <p>
+ *   This class is a concrete implementation of {@link com.microsoft.applicationinsights.internal.heartbeat.HeartBeatDefaultPayloadProviderInterface}
+ *   It enables setting Web-apps Metadata to heartbeat payload.
+ * </p>
+ *
+ * @author Dhaval Doshi
+ * @since 03-30-2018
+ */
+public class WebAppsDefaultHeartbeatProvider implements HeartBeatDefaultPayloadProviderInterface {
+
+  /**
+   * Name of the provider
+   */
+  private final String name = "webapps";
+
+  /**
+   * Collection holding default properties for this default provider.
+   */
+  private final Set<String> defaultFields;
+
+  /**
+   * Map for storing environment variables
+   */
+  private final Map<String, String> environmentMap;
+
+  /**
+   * Constructor that initializes fields and load environment variables
+   */
+  public WebAppsDefaultHeartbeatProvider() {
+    defaultFields = new HashSet<>();
+    environmentMap = System.getenv();
+    initializeDefaultFields(defaultFields);
+  }
+
+  @Override
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public boolean isKeyWord(String keyword) {
+    return defaultFields.contains(keyword);
+  }
+
+  @Override
+  public Callable<Boolean> setDefaultPayload(final List<String> disableFields,
+      final HeartBeatProviderInterface provider) {
+    return new Callable<Boolean>() {
+      volatile boolean hasSetValues = false;
+      volatile Set<String> enabledProperties = MiscUtils.except(defaultFields, disableFields);
+      @Override
+      public Boolean call() {
+        for (String fieldName : enabledProperties) {
+          try {
+            switch (fieldName) {
+              case "website_site_name":
```

Comment:
Do you read it for every heartbeat? In .NET SDK, Telemetry Initializers cached the value, and, therefore, were missing slot swap when the name changes. It looks like you read it every time here, so heartbeat will pick new name up.

Reply:
No, I am not reading it everytime, but this is a good point, if this is a scenario then we might have to read always or if not atleast every X minutes.

Reply:
See Microsoft/ApplicationInsights-dotnet-server#872 for the issue tracking the slot-swap problem.